MaxText/inference_microbenchmark_sweep.py

```diff
@@ -0,0 +1,153 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+"""Sweep across inference microbenchmarks."""
+
+import os
+import sys
+import json
+import jsonlines
+import inference_microbenchmark
+import max_utils
+import pyconfig
+from jax._src.lib import xla_extension
+
+
+def main():
+  """
+  User needs to set the config's inference_metadata_file, which is a path to a
+  json file.
+
+  This json should contain the following keys:
+    - key_value_axis_order_product_id_list: comma separated string of key_value_axis_order_product_id
+    - prefill_key_axis_order_list: comma delimited string of prefill_key_axis_order
+    - prefill_value_axis_order_list: comma delimited string of prefill_value_axis_order
+    - ar_key_axis_order_list: comma delimited string of ar_key_axis_order
+    - ar_value_axis_order_list: comma delimited string of ar_value_axis_order
+    - accelerator: name of the accelerator
+    - flatten_microbenchmark_results: Whether or not to flatten results. Should
+      be true
+  """
+  pyconfig.initialize(sys.argv)
+  config = pyconfig.config
+
+  with open(config.inference_metadata_file, encoding='utf-8') as json_file:
+    inference_metadata = json.load(json_file)
+    print(f"inference_metadata: {inference_metadata}")
```

Comment:
The inference_metadata in this case contains soly the 4 axis list right?

Reply:
It has 
```
  This json should contain the following keys:
    - key_value_axis_order_product_id_list: comma separated string of key_value_axis_order_product_id
    - prefill_key_axis_order_list: comma delimited string of prefill_key_axis_order
    - prefill_value_axis_order_list: comma delimited string of prefill_value_axis_order
    - ar_key_axis_order_list: comma delimited string of ar_key_axis_order
    - ar_value_axis_order_list: comma delimited string of ar_value_axis_order
    - accelerator: name of the accelerator
    - flatten_microbenchmark_results: Whether or not to flatten results. Should
      be true
```

I know `key_value_axis_order_product_id` is no longer used in the actual config/model. But I wanted to pass it in for the sake of logging:
```
    run_name = (
      f"{inference_metadata['accelerator']}-{config.model_name}-"
      f"{quant}-{key_value_axis_order_product_id}-{prefill_key_axis_order}-"
      f"{ar_key_axis_order}"
    )
```
Is that ok?

Reply:
Of course!